From 1ae141aac548c956a3a827e7543511b57579dbb9 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Tue, 6 Mar 2018 11:56:56 +0200
Subject: [PATCH 3427/5242] MLK-17689-1: drm:imx: dcss: Fix DCSS clock
 selection for MIPI

commit  c8a4128e9e6813057f462837e1db3ac23fe15445 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix the clock source selection for MIPI use-case.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-blkctl.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dcss/dcss-blkctl.c b/drivers/gpu/imx/dcss/dcss-blkctl.c
index 9607ce6..613c349 100644
--- a/drivers/gpu/imx/dcss/dcss-blkctl.c
+++ b/drivers/gpu/imx/dcss/dcss-blkctl.c
@@ -80,8 +80,13 @@ void dcss_blkctl_cfg(struct dcss_soc *dcss)
 {
 	struct dcss_blkctl_priv *blkctl = dcss->blkctl_priv;
 
-	dcss_writel((blkctl->clk_setting ^ HDMI_MIPI_CLK_SEL),
+	if (blkctl->hdmi_output)
+		dcss_writel((blkctl->clk_setting ^ HDMI_MIPI_CLK_SEL),
 		    blkctl->base_reg + DCSS_BLKCTL_CONTROL0);
+	else
+		dcss_writel((blkctl->clk_setting ^ HDMI_MIPI_CLK_SEL) |
+			    DISPMIX_PIXCLK_SEL,
+			    blkctl->base_reg + DCSS_BLKCTL_CONTROL0);
 
 	/* deassert clock domains resets */
 	dcss_blkctl_clk_reset(blkctl, 0, 0xffffff);
-- 
1.7.9.5

